src/NuGet.Core/NuGet.Packaging/NuspecReader.cs

```diff
@@ -639,12 +639,18 @@ private static List<string> GetFlags(string flags)
                 return EmptyList;
             }
 
-            var set = new HashSet<string>(
-                flags.Split(CommaArray, StringSplitOptions.RemoveEmptyEntries)
-                    .Select(flag => flag.Trim()),
-                StringComparer.OrdinalIgnoreCase);
+            // PERF: Avoid Linq on hot paths
+            var splitFlags = flags.Split(CommaArray, StringSplitOptions.RemoveEmptyEntries);
+            var set = new HashSet<string>(splitFlags.Length, StringComparer.OrdinalIgnoreCase);
+            for (int i = 0; i < splitFlags.Length; ++i)
```

Comment:
This can just be a foreach as C# will unroll under the covers to this.

Reply:
There are compile errors, so maybe this can be addressed in the same push @Erarndt

Reply:
Updated